recipe/2024-04-terraform-provider-secrets.md

```diff
@@ -0,0 +1,259 @@
+
+### User scenarios (optional)
+
+#### User story 1
+As an operator, I maintain a set of Terraform recipes for use with Radius. I have a set of provider configurations which are applicable across multiple recipes. These provider configurations now include secrets, which are injected into the Terraform configuration, allowing the Terraform process to access them as needed.
+
+
+## User Experience (if applicable)
+
+**Sample Input:**
+
+This is pulled in from existing document:
+[Design Document to support multiple Terraform Providers](https://github.com/radius-project/design-notes/pull/39/files)
+
+``` diff
+resource env 'Applications.Core/environments@2023-10-01-preview' = {
+  name: 'dsrp-resources-env-recipes-context-env'
+  location: 'global'
+  properties: {
+    compute: {
+      ...  
+    }
+    providers: {
+      ...
+    }
+    recipeConfig: {
+      terraform: {
+        authentication: {
+          ...
+        }
+        providers: {
+          azurerm: [
+            {
+              subscriptionid: 1234
+              tenant_id: '745fg88bf-86f1-41af'
++              secrets: { // Individual Secrets from SecretStore
++                my_secret_1: {
++                  source: secretStoreConfig.id
++                  key: 'secret.one'
++                }
++                my_secret_2: {
++                  source: secretStoreConfig.id
++                  key: 'secret.two'
++                }
++              }
+            }, {
+              subscriptionid: 1234
+              tenant_id: '745fg88bf-86f1-41af'
+              alias: 'az-paymentservice'
+            } ]
+          gcp: [
+            {
+              project: 1234
+              regions: [ 'us-east1', 'us-west1' ]
+            }
+          ]
+          oraclepass: [
+            {
+              database_endpoint: '...'
+              java_endpoint: '...'
+              mysql_endpoint: '...'
+            }
+          ]
+        }
+      }
+      env: {
+        MY_ENV_VAR_1: 'my_value'
+      }
++      envSecrets: { // Individual Secrets from SecretStore
++        MY_ENV_VAR_2: {
++          source: secretStoreConfig.id
++          key: 'envsecret.one'
++        }
++        MY_ENV_VAR_3: {
++          source: secretStoreConfig.id
++          key: 'envsecret.two'
++        }
++     }
+    }
+    recipes: {
+    }
+  }
+}
+
+```
+
+**Sample Output:**
+n/a
+
+**Sample Recipe Contract:**
+n/a
+
+## Design
+
+### High Level Design
+
+The design for handling secrets for Private Terraform modules established the basis for data flow between Engine and Driver components in the codebase. For Terraform Providers, we will continue to maintain the flow where secrets are retrieved in the Engine and passed into the Driver in the Execute() call. 
+The following diagram describes the updated data flow. The calls in green highlight updates to existing code.
+
+
+![alt text](2024-04-terraform-provider-secrets.png)
+
```

Comment:
Love the diagram. BTW if it's easier in the future you can embed mermaid diagrams in markdown. Sequence diagram support is really good.

https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/